src/test/java/io/strimzi/kafka/bridge/mqtt/mapper/MqttKafkaRegexMapperTest.java

```diff
@@ -0,0 +1,245 @@
+                mapper.map("building/4/room/23/temperature").kafkaTopic(), is("building_4_room_23"));
+
+        assertThat("The key for building/([^/]+)/room/(\\d{1,3}).* should be expanded to building_$1",
+                mapper.map("building/4/room/23/temperature").kafkaKey(), is("building_4"));
+
+        // Test for building/([^/]+).* pattern
+
+        assertThat("Mqtt pattern building/([^/]+).* should be mapped to building_$1_others",
+                mapper.map("building/405/room").kafkaTopic(), is("building_405_others"));
+
+        assertThat("The key for building/([^/]+).* should be expanded to building_$1",
+                mapper.map("building/405/room").kafkaKey(), not("4"));
+
+        // Test for building.* pattern
+        assertThat("Mqtt pattern building.* will be mapped to building_101_others because building/([^/]+).* was defined before building.*",
+                mapper.map("building/101").kafkaTopic(), not("building_others"));
+
+        assertThat("building.* should be mapped to building_others",
+                mapper.map("building").kafkaTopic(), is("building_others"));
+
+        // Test for fleet/([0-9])/vehicle/(\\w+).* pattern
+        assertThat("Mqtt pattern fleet/([0-9])/vehicle/(\\w+).* should be mapped to fleet_$1",
+                mapper.map("fleet/9/vehicle/23/velocity").kafkaTopic(), is("fleet_9"));
+
+        assertThat("The key for fleet/([0-9])/vehicle/(\\w+).* should be expanded to fleet$1",
+                mapper.map("fleet/9/vehicle/23/velocity").kafkaKey(), is("fleet9"));
+
+        // Test for sensor.* pattern
+        assertThat("Mqtt pattern sensor.* should be mapped to sensor_data",
+                mapper.map("sensors/temperature/data").kafkaTopic(), is("sensor_data"));
+
+        assertThat("The key for sensor.* should be expanded to sensor",
+                mapper.map("sensors/temperature/data").kafkaKey(), is("sensor"));
+
+        // Test for sport/tennis/(\\w+).* pattern
+        assertThat("Mqtt pattern sport/tennis/(\\w+).* should be mapped to sports_$1",
+                mapper.map("sport/tennis/player1").kafkaTopic(), is("sports_player1"));
+
+        assertThat("Mqtt pattern sport/tennis/(\\w+).* should be mapped to sports_$1",
+                mapper.map("sport/tennis/player100/ranking").kafkaTopic(), is("sports_player100"));
+
+        assertThat("Mqtt pattern sport/tennis/(\\w+).* should be mapped to sports_$1",
+                mapper.map("sport/tennis/player123/score/wimbledon").kafkaTopic(), is("sports_player123"));
+
+        // Test for ([^/]+)/recipes.* pattern
+        assertThat("Mqtt pattern ([^/]+)/recipes.* should be mapped to my_recipes",
+                mapper.map("angolan/recipes/caculu/fish").kafkaTopic(), is("angolan_recipes"));
+
+        assertThat("The key for ([^/]+)/recipes.* should be expanded to $1",
+                mapper.map("angolan/recipes/caculu/fish").kafkaKey(), is("angolan"));
```

Comment:
Usual note. Calling `map` once for each pattern and verify topic and key on the resulting `MappingResult`.